Problem: upgrade to siwe 0.5 fails

```
error[E0061]: this function takes 2 arguments but 4 arguments were supplied
[826](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:827)
   --> common/src/login.rs:40:31
[827](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:828)
    |
[828](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:829)
40  |         let result = self.msg.verify(sig, None, None, None);
[829](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:830)
    |                               ^^^^^^ ---  ----  ----  ---- argument of type `std::option::Option<_>` unexpected
[830](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:831)
    |                                      |    |     |
[831](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:832)
    |                                      |    |     argument of type `std::option::Option<_>` unexpected
[832](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:833)
    |                                      |    expected `&siwe::VerificationOpts`, found enum `std::option::Option`
[833](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:834)
    |                                      expected `&[u8]`, found array `[u8; 65]`
[834](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:835)
    |
[835](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:836)
    = note: expected reference `&siwe::VerificationOpts`
[836](https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3271952032/jobs/5382364837#step:5:837)
                    found enum `std::option::Option<_>`
```

--- a/bindings/cpp/Cargo.toml
+++ b/bindings/cpp/Cargo.toml
@@ -13,7 +13,7 @@ cxx = "1"
 anyhow = "1"
 serde="1"
 serde_json="1"
-siwe = { version = "0.4" }
+siwe = { version = "0.5" }
 ethers = { version = "0.17", features = ["rustls"] }
 hex = "0.4"
 tokio = { version = "1", features = ["rt"] }

--- a/bindings/cpp/src/lib.rs
+++ b/bindings/cpp/src/lib.rs
@@ -535,7 +535,7 @@ pub mod ffi {
         /// NOTE: the server may still need to do extra verifications according to its needs
         /// (e.g. verify chain-id, nonce, uri + possibly fetch additional data associated
         /// with the given Ethereum address, such as ERC-20/ERC-721/ERC-1155 asset ownership)
-        fn verify_logininfo(self: &CppLoginInfo, signature: &[u8]) -> Result<Vec<u8>>;
+        fn verify_logininfo(self: &CppLoginInfo, signature: &[u8]) -> Result<()>;
 
         /// create cronos tx info to sign
         pub fn new_eth_tx_info() -> EthTxInfoRaw;
@@ -1056,12 +1056,7 @@ impl CppLoginInfo {
     /// (as per EIP-4361). The returned vector is a serialized recoverable signature
     /// (as used in Ethereum).
     pub fn sign_logininfo(&self, private_key: &PrivateKey) -> anyhow::Result<Vec<u8>> {
-        let message = self.logininfo.msg.to_string();
-        let secretkey = private_key.key.clone();
-        let ret = secretkey
-            .eth_sign(message.as_bytes(), self.logininfo.msg.chain_id)
-            .map(|x| x.to_vec())?;
-        Ok(ret)
+        Ok(self.logininfo.sign(&private_key.key)?)
     }
 
     /// Verify Login Info
@@ -1072,13 +1067,11 @@ impl CppLoginInfo {
     /// NOTE: the server may still need to do extra verifications according to its needs
     /// (e.g. verify chain-id, nonce, uri + possibly fetch additional data associated
     /// with the given Ethereum address, such as ERC-20/ERC-721/ERC-1155 asset ownership)
-    pub fn verify_logininfo(&self, signature: &[u8]) -> anyhow::Result<Vec<u8>> {
-        let sig: [u8; 65] = signature
-            .try_into()
-            .map_err(|_e| EthError::SignatureError)?;
+    pub fn verify_logininfo(&self, signature: &[u8]) -> anyhow::Result<()> {
+        // TODO Reuse runtime on blocking function
+        let rt = tokio::runtime::Runtime::new()?;
         // FIXME: domain, nonce, timestamp
-        let result = self.logininfo.msg.verify(sig, None, None, None)?;
-        Ok(result)
+        Ok(rt.block_on(self.logininfo.verify(signature))?)
     }
 }
 fn convert_amount(

--- a/common/Cargo.toml
+++ b/common/Cargo.toml
@@ -41,7 +41,7 @@ secrecy = "0.8"
 serde = "1"
 serde_json = "1"
 serde_with = "2"
-siwe = { version = "0.4", optional = true }
+siwe = { version = "0.5", optional = true }
 tendermint = "0.23"
 tendermint-proto = "0.23"
 tendermint-rpc = "0.23"

--- a/common/src/login.rs
+++ b/common/src/login.rs
@@ -1,5 +1,5 @@
 use crate::{EthError, SecretKey};
-use siwe::Message;
+use siwe::{Message, VerificationOpts};
 use std::fmt::Display;
 
 /// The wrapper structure that contains
@@ -32,12 +32,10 @@ impl LoginInfo {
     /// NOTE: the server may still need to do extra verifications according to its needs
     /// (e.g. verify chain-id, nonce, uri + possibly fetch additional data associated
     /// with the given Ethereum address, such as ERC-20/ERC-721/ERC-1155 asset ownership)
-    pub fn verify(&self, signature: &[u8]) -> Result<(), EthError> {
-        let sig: [u8; 65] = signature
-            .try_into()
-            .map_err(|_e| EthError::SignatureError)?;
+    pub async fn verify(&self, signature: &[u8]) -> Result<(), EthError> {
         // FIXME: domain, nonce, timestamp to be passed
-        let result = self.msg.verify(sig, None, None, None);
+        let opts = VerificationOpts::default();
+        let result = self.msg.verify(signature, &opts).await;
         result.map_err(|_e| EthError::SignatureError).map(|_| ())
     }
 }
@@ -91,15 +89,15 @@ mod tests {
         LoginInfo { msg }
     }
 
-    #[test]
-    pub fn test_sign() {
+    #[tokio::test]
+    pub async fn test_sign() {
         let wallet = SecretKey::default();
         let login_info = get_logininfo(Some(&wallet));
 
         let signature = login_info.sign(&wallet);
         assert!(signature.is_ok());
         let sig = signature.unwrap();
-        assert!(login_info.verify(&sig).is_ok());
+        assert!(login_info.verify(&sig).await.is_ok());
     }
 
     #[test]

